### PR TITLE
Amqp connector option for setting sender/receiver capabilities

### PR DESCRIPTION
**Pull request by ozangunalp**
Adds "capabilities" configuration for both incoming and outgoing channels.

### CODE REVIEW

File: smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpConnector.java [hunk 1]
```diff
@@ -295,6 +310,14 @@ void init() {
                 .ignore();
     }
 
+    public List<String> getClientCapabilities(AmqpConnectorCommonConfiguration configuration) {
+        if (configuration.getCapabilities().isPresent()) {
+            String capabilities = configuration.getCapabilities().get();
+            return Arrays.stream(capabilities.split(",")).map(String::trim).collect(Collectors.toList());
```
Comment (cescoffier) [inline smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpConnector.java:316]: Probably best to filter out blank strings.

**Review by cescoffier — CHANGES_REQUESTED** [review]
Looks great, I just made a very small "defensive" comment.

**Comment by cescoffier** [comment]
Do not forget to run `mvn clean install -DskipTests` from the project root to be sure to have the up do date connector configuration tables in the documentation (that's what the up_to_date_antora is reporting).

File: smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpConnector.java [hunk 2]
```diff
@@ -295,6 +310,16 @@ void init() {
                 .ignore();
     }
 
+    public List<String> getClientCapabilities(AmqpConnectorCommonConfiguration configuration) {
+        if (configuration.getCapabilities().isPresent()) {
+            String capabilities = configuration.getCapabilities().get();
+            return Arrays.stream(capabilities.split(","))
+                    .filter(s -> !s.isEmpty())
+                    .map(String::trim).collect(Collectors.toList());
```
Comment (cescoffier) [inline smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpConnector.java:318]: you should invert the two statements as you would not filter out " "

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpConnector.java (hunk 2 above)
Comment (ozangunalp) [reply smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpConnector.java:318]: That's right!

**Review by cescoffier — APPROVED** [review]

**Pull request merged**